rust/src/lib/unit_tests/ifaces.rs

```diff
@@ -426,3 +426,303 @@ fn test_unknown_iface_type() {
         "foo_type\n"
     );
 }
+
+#[test]
+fn test_hsr_mac_sync_no_mac_addresses_defined() {
+    let mut cur_ifaces = Interfaces::new();
+
+    let mut iface_with_mac = new_eth_iface("eth1");
+    iface_with_mac.base_iface_mut().mac_address =
+        Some("AA:BB:CC:DD:EE:FF".to_string());
+    cur_ifaces.push(iface_with_mac);
+    cur_ifaces.push(new_eth_iface("eth2"));
+
+    let mut des_ifaces = Interfaces::new();
+    des_ifaces.push(new_hsr_iface("hsr0", "eth1", "eth2"));
+    des_ifaces.push(new_eth_iface("eth1"));
+    des_ifaces.push(new_eth_iface("eth2"));
+
+    let merged_ifaces = MergedInterfaces::new(
+        des_ifaces,
+        cur_ifaces,
+        crate::NetworkStateMode::Apply,
+        false,
+    )
+    .unwrap();
+
+    let hsr_iface =
+        merged_ifaces.get_iface("hsr0", InterfaceType::Hsr).unwrap();
+    assert_eq!(
+        hsr_iface
+            .for_apply
+            .as_ref()
+            .unwrap()
+            .base_iface()
+            .mac_address,
```

Comment:
Create a `_get_mac` helper function so you can do more readable asserts: `assert_eq!(_get_mac(hsr_iface), "AA:BB:CC:DD:EE:FF")`, here and in the rest of the tests.